### PR TITLE
Baseline experiments 

### LINKED ISSUE #18: Baseline experiments
This issue describes baseline experiments for the RSNA 2024 kaggle data challenge.
The repo associated can be found [here](https://github.com/ivadomed/lumbar-classification-rsna-challenge-2024/tree/baseline-experiments).

The baseline strategy is <b>One condition, one level, one model</b>. 
It's training a [ResNet18](https://docs.monai.io/en/stable/networks.html#resnet) model, implemented by Monai.
Hyperparameters and preprocessing parameters can be modified via the `config.json` file.

ToDo : 

- [ ] Decide a strategy for resampling minority classes (we're in an unbalanced setting)

### LINKED ISSUE #25: Choice of patch extraction method
Given centroids corresponding to each disk labeling,  we want to extract patches around them. 
Thus, we need to chose a patch extraction method. 

We can either chose boxes with sides parallel to image axes but it's hard use because we could lose some relevent information if boxes are narrow or have too much overlap between patches if boxes are too large. 

I propose to design boxes with axes following spine curvature, see the illustration below :

Capture d’écran du 2024-08-12 09-44-36

Capture d’écran du 2024-08-12 09-47-56

If we chose the method "following spine curvature", it requires some postprocessing step like rotation to extract the patches.

Capture d’écran du 2024-08-18 20-18-17

### LINKED ISSUE #27: TotalSpineSeg for context extraction
Following a two stages strategies ( 1) disc labeling and 2) grading network) I tried [TotalSpineSeg](https://github.com/neuropoly/totalspineseg/releases/tag/r20240826) for disc labeling on RSNA dataset. 

Results are impressive (see images below) :

Capture d’écran du 2024-09-06 13-50-56
Capture d’écran du 2024-09-06 13-51-01
Capture d’écran du 2024-09-06 13-51-15

Then, we can use segmentation mask to extract regions of interest  and take them as input of a NN : 

Capture d’écran du 2024-09-10 13-33-36

### PR DESCRIPTION
**Pull request by simonqueric**
First implementation of baseline for the RSNA 2024 kaggle data challenge. 

What has been done so far : 

- [x] `dataset.py` Dataset class assuming a BIDS data structure, returning a dictionary with keys "image", "label", "study_id"
- [x] `config.json` Config file with hyperparameters and preprocessing parameters 
- [x] `train_and_test.py` Training loop 

Todo : 

- [ ] Add a resampling strategy for minority classes (Moderate and Severe classes)
- [ ] Change `config.json` so that preprocessing parameters (voxel spacing, shape of volumes) depend on the contrast 

Fixes #18

### CODE REVIEW

File: train_and_test.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    seq2cond = {"sag-T1" : ["left_neural_foraminal_narrowing", 
+                            "right_neural_foraminal_narrowing"],
+                        
+                "sag-T2" : ["spinal_canal_stenosis"], 
+                
+                "ax-T2" : ["left_subarticular_stenosis", 
+                            "right_subarticular_stenosis"]  
+                }
+    
+    text2int = {"Normal/Mild" : 0, "Moderate" : 1, "Severe" : 2}
```
Comment (jcohenadad) [inline train_and_test.py]:
```suggestion
    text2int = {"Normal/Mild": 0, "Moderate": 1, "Severe": 2}
```
PEP8 convention

File: notebook/totalspineseg.ipynb
Comment (valosekj) [inline notebook/totalspineseg.ipynb:1]: nice notebook!
